hw/top_earlgrey/rdc/rdc_waivers.tcl

```diff
@@ -169,3 +169,8 @@ set_rule_status -rule E_RST_METASTABILITY -status Waived \
   -expression {(SourceReset=="POR_N") && \
     (ResetFlop=~"u_ast.u_no_scan_poks*") && \
     (MetaStableFlop=~"u_ast.u_no_scan_por_sync*")}
+
+# W_ASYNC_RST_FLOPS
+set_rule_status -rule W_ASYNC_RST_FLOPS -status Waived \
```

Comment:
hmmm... could you let me us know what the path is? because during TEST / RMA, we are allowed to sample the tap_strap live.

Reply:
It connects to TRST_N. So, as long as TRST_N is asserted while capturing the tap_strap, it is fine. Can we have that assumption?

Reply:
hmmmm... does it have to be asserted? Or can it just..static? Ie 0 or 1 all the time?

Reply:
Since the tap_strap_q behaves as a reset, I think it doesn't matter. But as strap is 2-bit signal, keeping TRST_N to 0 is preferred.

Reply:
hmmm... we may need to look at this then. Because I think we do expect Nuvoton to change the strap values live during TEST state.  During test state at least, these values aren't really "straps" anymore, they are live selections.

So is the tool basically saying during those states, the live selection may cause a reset glitch?

@arnonsha do you happen to know, in the worst case, is it okay to always assert TRST_N while doing a switch in your case? I'm guessing if you need to switch between the tool inserted one and rv_dm that might not be good, because the reset could wipe out your tap settings.  

@eunchan we could guarantee that during the switch there won't be any live transactions though...

Reply:
so at the moment, all unselected TAPs will automatically be held in reset (TRST_N asserted). so in that regard, pulling TRST_N before and after switching is not much different from the current behavior. if this is not expected, we need to change the following lines:

https://github.com/lowRISC/opentitan/blob/587d8592405d80c64cc7f3e6f173ff60442ec128/hw/ip/pinmux/rtl/pinmux_strap_sampling.sv#L311-L315

@eunchan the `tap_strap` selector is currently driven from a flop. would it be better if we decoded the selection into one-hot before capturing it in a flop, and then drive the TAP demux with the flopped one-hot signals in order to avoid glitches?

Reply:
> @eunchan the `tap_strap` selector is currently driven from a flop. would it be better if we decoded the selection into one-hot before capturing it in a flop, and then drive the TAP demux with the flopped one-hot signals in order to avoid glitches?
>
I think it depends. When tap value is changed, multi-bit will be changed even in one-hot encoding case. It will create glitches. If we want to be safe, the each reset  should be latched. However, this means we need the TCK clock for the reset.

So, the best way I think is to keep TRST_N to 0 before changing the strap.

Reply:
yeah i agree with @eunchan . I think what he's describing (the multi-bit) was what I assumed the tool is complaining about also.   But on that note, I kind of forgot that when switching away from an interface its held in reset.

We might want to double check this with @arnonsha . I dont think they intend to do rv_dm -> dft -> rv_dm, but if they were to to do that, everything the dft interface sets up could be wiped out.

Reply:
I do see a scenario where we do rv_dm -> dft . I currently don't see a scenario where we go back to rv_dm from dft, however I don't want to rule out such option.
@tjaychen , It's been a while, but I think the debugger will reset/program the TAP once again regardless of it's condition when it reconnects to it.

Reply:
Thanks @arnonsha So, if JTAG asserts the reset every time it connects, I think this waiver is fine.

Reply:
that sounds good!
I just wanted to make sure we don't rely on some registers being programmed and to remain available later after switching.

Reply:
Wait.. I thought the TRST_N does not reset the dm_csrs module. Let me double check.

Reply:
sorry i'm not referring to `rv_dm`. I was talking about the tool inserted tap and any of those control registers.
I'm fairly certain those are completely on `TCK` and `TRST_N` and would just get wiped out together.

Reply:
Hum.. then if somehow the internal registers need to be alive throughout attaching and detaching, does it mean that we need to keep TRST_N when the TAP is not selected? 

If we need that case, I think only viable option is for each TAP to have their own TRST_N unless we make the selection in cyclic manner (RV_DM -> LC_CTRL -> DFT -> RV_DM) to avoid glitches. We do not have PADs for the additional TRST_Ns so, we need to change how to select the TAP.

Reply:
yeah that's right. But sounds like we don't need the persistence though, according to what @arnonsha is saying.

Reply:
Yeah if we don't need persistence, I think we are good.

@eunchan I think TRST only resets the response FIFO in dm_csrs, but not the rest.